spring-context/src/main/java/io/micronaut/spring/context/MicronautApplicationContext.java

```diff
@@ -380,6 +386,11 @@ public void addApplicationListener(ApplicationListener<?> listener) {
         beanFactory.getBeanContext().registerSingleton(listener);
     }
 
+    @Override
+    public void removeApplicationListener(ApplicationListener<?> listener) {
+        beanFactory.getBeanContext().destroyBean(listener);
```

Comment:
Is this the correct way to remove the listener singleton?

Reply:
looks ok